static/js/components/ClassificationForm.jsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect, useReducer } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useForm, Controller } from 'react-hook-form';
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import Input from "@material-ui/core/Input";
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import FormControl from "@material-ui/core/FormControl";
+import MenuItem from "@material-ui/core/MenuItem";
+import { makeStyles } from "@material-ui/core/styles";
+import Autocomplete, { createFilterOptions } from '@material-ui/lab/Autocomplete';
+import * as Actions from '../ducks/source';
+
+
+function makeMenuItem(taxonomy, index) {
+
+  const render_string = `[${taxonomy.id}] ${taxonomy.name} ${taxonomy.version}`;
+
+  return (
+    <MenuItem value={index} key={index.toString()}>
+      {render_string}
+    </MenuItem>
+  );
+}
+
+
+const ClassificationForm = ({ obj_id, taxonomyList }) => {
+
+  function reducer(state, action) {
+      switch (action.name) {
+        case 'taxonomy_index':
+          return {
+            ...state,
+            [action.name]: action.value,
+            allowed_classes: taxonomyList[action.value].allowed_classes,
+          };
+        default:
+          return {
+            ...state,
+            [action.name]: action.value
+          };
+      }
+  }
+
+  const submitDispatch = useDispatch();
+
+  const initialState = {
+    taxonomy_index: taxonomyList.length > 0 ? 0 : null,
+    classification: null,
+    probability: 1.0,
+    class_select_enabled: false,
+    probability_select_enabled: false,
+    probability_errored: false,
+    allowed_classes: taxonomyList.length > 0 ? taxonomyList[0].allowed_classes : [null]
+  }
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { handleSubmit, getValues, reset, register, control } = useForm();
+
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      fullWidth: true,
+      display: 'flex',
+      wrap: 'nowrap'
+    }
+  }));
+  const classes = useStyles();
+
+  if (taxonomyList.length === 0) {
+    return (
+      <b>
+        No taxonomies loaded...
+      </b>
+    );
+  }
+
+  const handleTaxonomyChange = (event) => {
+    dispatch({name: "taxonomy_index", value: event.target.value});
+    dispatch({name: "classification", value: ""});
+    dispatch({name: "class_select_enabled", value: true});
+    dispatch({name: "probability_select_enabled", value: false});
+    dispatch({name: "probability_errored", value: false});
+    dispatch({name: "probability", value: 1.0});
+  };
+
+  const handleClasschange = (event, value) => {
+    dispatch({name: "classification", value: value});
+    dispatch({name: "probability_select_enabled", value: true});
+  };
+
+  const processProb = (event, value) => {
```

Comment:
When I classify something within a single taxonomy with P=0.9, then classify it as something else within the same taxonomy with P=0.1, I'd expect both classifications to be displayed with their corresponding probabilities. If I have a total probability > 1, I'd expect an error to be thrown. Instead, what happens is the first classification (and its probability) are just overwritten by the second, so the only way the probabilities will sum to one is if I classify something with P=1. If only one classification can be supported at a time per taxonomy, the Probability selector is not really being used.

Reply:
No classifications are overwritten (the most recent one is shown near the top of the Source page) and by design I think it's ok for classifications not to sum to 1. For example, if new data arrives or another ML model with the same token wants to write a classification statement which may be in conflict with the first one. If you set P=0 then that classification will not be shown in the summary at top.

I thought about having classification groupings which would allow someone to add a classification vector with the same tag/group. But for now all the classes are atomized and distinct from each other (for accounting purposes).